### ISSUE TITLE
Console window strategy?

### ISSUE THREAD

**Issue by aolo2** [nakst/gf#25]
Right now, the console window is not like the others: not only does it serve its primary purpose (input-output from gdb), but it has the status square, the `Menu` button, and the `Donate` button (questionable desicion to put it there, IMHO).

Currently, the console window is a "last resort" way to tell gdb to do something, when the usual interface and shortcuts are not able to it. For me personally, there are only two cases for using it: placing a breakpoint by function name, and using format specifiers to print a value in hex/binary etc.

Ok, I've lost my train of thought a bit.. Here are my main points:
1. I think the menu button and the status should be a separate pane
2. If there are any things you often find yourself doing via the console, it might be time to move them to some separate UI (but not always. For one, a quick `print arr[123]` is, in my opinion, cleaner than adding the value to the watch window only for it to be left there as garbage)

To sum it up, I guess I'm asking what your thoughts are on the future of the Console window. Is it a "last resort" for things not in the UI or something else?

**Comment by nakst** [comment]
`gf` was designed to augment GDB, rather than replace it. Because of that, the console is an important part of the UI (at least for me). If you don't like it, you can always modify the layout to make the console smaller by default, or hide it away in a tab (or remove it all together). 

> For one, a quick print arr[123] is, in my opinion, cleaner than adding the value to the watch window only for it to be left there as garbage)

This is a key point for me. There are some things that the console is already really good at! Adding separate UI for each individual task would complicate the application.

I might consider adding a separate window that contains just the traffic light and menu button in the future, but I not sure if this is the right direction.

Side note: I decided to move the Donate button into the menu. I wanted to have it in a prominent place in the UI similar to what WizTree does, but having it in the tab traversal order was bothersome.

**Comment by aolo2** [comment]
I see what you mean and mostly agree. By the way, removing Console from the layout segfaults gf on start. It's a whole other question why one would need to remove it, but it's a bug nonetheless.

Considering this issue, I'm not sure what to do with it.

**Comment by nakst** [comment]
> removing Console from the layout segfaults gf on start
Could you send your layout string?

**Comment by aolo2** [comment]
> > removing Console from the layout segfaults gf on start
> 
> Could you send your layout string?

```
[ui]                                                                                                            
font_size=12                                                                                                    
layout=h(60,v(80,Source),v(80,t(Watch,Registers,Data),t(Stack,Breakpoints,Files,Commands)))
```

**Comment by nakst** [comment]
The part `v(80,Source)` is incorrect; the vertical split must have exactly 2 children. I've improved the error messages here, and updated the documentation. If you replace it with just `Source`, the layout should work.

I'm going to close this issue. If you have any specific feature requests, make a new issue for them. With regards to the console window itself, I have no plans to change its purpose.
